File: Hydrogene/Toffee/UIKit/UINavigationController/UINavigationController/RootViewController.cs
namespace UINavigationController
{
	using UIKit;

	[IBObject]
	class RootViewController : UIViewController
	{
		public override id init()
		{
			this = base.initWithNibName("RootViewController") bundle(null);
			if (this != null)
			{
				title = "UINavigationController";
			}
			return this;
		}

		[IBAction]
    public void buttonPressed(id sender)
    {
        SecondViewController secondViewController = new SecondViewController();
        this.navigationController.pushViewController(secondViewController) animated(true); 
	  }
  }
}

File: Oxygene/Toffee/UIKit/UINavigationController/UINavigationController/RootViewController.pas
namespace UINavigationController;

interface

uses
  UIKit;

type
  [IBObject]
  RootViewController = public class(UIViewController)
  private
  public
    method init: id; override;
    [IBAction] method buttonPressed(sender :id);
  end;

implementation

method RootViewController.init: id;
begin
  self := inherited initWithNibName('RootViewController') bundle(nil);
  if assigned(self) then begin
    title := 'UINavigationController';
  end;
  result := self;
end;

method RootViewController.buttonPressed(sender: id);
begin
  var secondViewController := new SecondViewController();
  self.navigationController.pushViewController(secondViewController) animated(true);
end;


end.
